From 5dba112470ff48982ec020fe00ab07a93fb58ab1 Mon Sep 17 00:00:00 2001
From: Paul Gortmaker <paul.gortmaker@windriver.com>
Date: Fri, 26 Mar 2010 12:09:36 -0400
Subject: [PATCH] vdk_port: add struct PortConfig

This was originally in vdk_interface.h but it clearly is a
part of the port infrastructure.

Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>
---
 include/vbi/vdk_interface.h |    1 -
 include/vbi/vdk_port.h      |   25 +++++++++++++++++++++++++
 2 files changed, 25 insertions(+), 1 deletions(-)

diff --git a/include/vbi/vdk_interface.h b/include/vbi/vdk_interface.h
index 021b431..f417a17 100644
--- a/include/vbi/vdk_interface.h
+++ b/include/vbi/vdk_interface.h
@@ -160,7 +160,6 @@
 typedef char VB_WRHV_NAME[VB_MAX_WRHV_NAME_LENGTH];
 
 #define VB_MAX_EXCEPTION_TYPES	32
-#define MAX_NAME_LENGTH 80
 
 struct vb_log_cfg
 {
diff --git a/include/vbi/vdk_port.h b/include/vbi/vdk_port.h
index 96e225a..9a28cc9 100644
--- a/include/vbi/vdk_port.h
+++ b/include/vbi/vdk_port.h
@@ -47,6 +47,31 @@ typedef enum                            /* queuing port protocol     */
 #define PORT_Q_AVAILABLE			4
 #define PORT_Q_DEST_UNAVAILABLE			9
 
+#define MAX_NAME_LENGTH 80
+
+struct portConfig
+{
+	unsigned long id;		/* unique port ID                   */
+	unsigned char *buf;		/* port payload buffer vaddr        */
+	unsigned long bufAttr;		/* port payload buffer attribute    */
+	char name[MAX_NAME_LENGTH];	/* port name                        */
+	unsigned long vbId;		/* parent vb                        */
+	unsigned long moduleId;		/* parent module                    */
+	unsigned long inChannel;	/* in channel                       */
+	unsigned long outChannel;	/* out channel                      */
+	PORT_DIRECTION_TYPE direction;	/* SOURCE or DESTINATION            */
+	PORT_MODE_TYPE mode;		/* QUEUING or SAMPLING              */
+	PORT_PROTOCOL_TYPE protocol;	/* RECEIVER_DISCARD or SENDER_BLOCK */
+	unsigned long msgSize;		/* max. message size (PAGE_ALIGNED) */
+	unsigned long numMsgs;		/* max. capacity for queuing ports  */
+};
+
+struct partPortConfig
+{
+	unsigned long numPorts;
+	struct portConfig portInfo[];
+};
+
 struct portStatus				/* port runtime status        */
 {
 	unsigned char *sBuf;			/* SIPC buffer                */
-- 
1.7.0

